### DISCUSSION TITLE
call GO function when closing application window

### DISCUSSION THREAD

**Discussion by Juniornewxt** [wailsapp/wails#2958] [General]
Friends, I created a program here that uses some temporary files in its execution, I would like to make it so that when I click on close the window, these files are cleaned, is it possible to do this? Do you have any example of how you could call functions in Go when you close the window?

**Comment by ianmjones** [comment]
One of the [app options](https://wails.io/docs/reference/options) is `OnBeforeClose`.

You give it a function, such as the [beforeClose example](https://wails.io/docs/reference/options#onbeforeclose), that will run just as the window is about to be closed.

Depending on the template you used to create your Wails app, you might already have a function already hooked up to `OnBeforeClose`.

In a similar vein, there's an [OnShutdown](https://wails.io/docs/reference/options#onshutdown) function option that is called after the frontend has been destroyed and the application is about to fully quit, which might be more apt if just doing some silent cleanup.

**Reply by Juniornewxt** [reply]
Thank you, it worked perfectly.

**Comment by Juniornewxt** [comment]
Thank you, it worked perfectly.

Remember to put it in main.go
OnBeforeClose: app.beforeClose,
